src/database_migration.py

```diff
@@ -0,0 +1,142 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Provide the DatabaseMigration class to manage database migrations."""
+
+import logging
+from typing import Literal, cast
+
+import ops
+from ops.pebble import ExecError
+
+from constants import FLASK_STATE_DIR
+from exceptions import CharmConfigInvalidError
+
+logger = logging.getLogger(__name__)
+
+
+DatabaseMigrationStatus = Literal["PENDING", "COMPLETED", "FAILED"]
+
+
+class DatabaseMigration:
+    """The DatabaseMigration class that manages database migrations.
+
+    Attrs:
+        script: the database migration script.
+        COMPLETED: one of the possible database migration status.
+        FAILED: one of the possible database migration status.
+        PENDING: one of the possible database migration status.
+    """
+
+    _STATUS_FILE = FLASK_STATE_DIR / "database-migration-status"
+    _COMPLETED_SCRIPT_FILE = FLASK_STATE_DIR / "completed-database-migration"
+
+    COMPLETED: DatabaseMigrationStatus = "COMPLETED"
+    FAILED: DatabaseMigrationStatus = "FAILED"
+    PENDING: DatabaseMigrationStatus = "PENDING"
+
+    def __init__(self, flask_container: ops.Container, database_migration_script: str | None):
+        """Initialize the DatabaseMigration instance.
+
+        Args:
+            flask_container: The flask container object.
+            database_migration_script: The database_migration_script configuration.
+        """
+        self._container = flask_container
+        self._script = database_migration_script
+
+    @property
+    def script(self) -> str | None:
+        """Get the database migration script."""
+        return self._script
+
+    def get_status(self) -> DatabaseMigrationStatus:
+        """Get the database migration run status.
+
+        Returns:
+            One of "PENDING", "COMPLETED", or "FAILED".
+        """
+        return cast(
```

Comment:
Would it be clearer to split this up into a few lines to first handle the file doesn't exist case and then read?